### PR TITLE
Add lifecycle extension to core, migrate to AndroidX

### PR DESCRIPTION
**Pull request by spoorthipujariadobe**
## Description

1. All files are just a copy from lifecycle extension to core project.
2. Modified build.gradle file to use Androidx dependency and migrate project to AndroidX.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/android-lifecycle-library/build.gradle [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+}
+
+dependencies {
+    implementation 'androidx.appcompat:appcompat:1.0.0'
+    implementation project(path: ':android-core-library')
+    // unit tests
+    testImplementation "junit:junit:4.13.2"
+    testImplementation project(path: ':android-lifecycle-library')
```
Comment (yangyansong-adbe) [inline code/android-lifecycle-library/build.gradle:54]: Now we can remove this "testImplementation", it was a workaround for fixing a bug of the latest Android Studio.

File: code/android-lifecycle-library/build.gradle [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    //noinspection GradleDependency
+    testImplementation "org.mockito:mockito-core:2.22.0"
+    testImplementation 'org.powermock:powermock-api-mockito2:2.0.0'
```
Comment (yangyansong-adbe) [inline code/android-lifecycle-library/build.gradle:57]: I don't think we used power-mock in Lifecycle's test code, can we remove it?

**Review by yangyansong-adbe — APPROVED** [review]
LGTM, only a couple of small comments.

File: code/android-lifecycle-library/build.gradle (hunk 2 above)
Comment (spoorthipujariadobe) [reply code/android-lifecycle-library/build.gradle:57]: Thought we might need it to mock MobileCore or Platform services. Will remove once I fix tests

**Comment by spoorthipujariadobe** [comment]
#128

**Pull request merged**